backend/src/data/repositories/user-repository.ts

```diff
@@ -1,11 +1,34 @@
-import { UserModel } from '../models';
+import { UserModel, DoctorModel, ClinicModel } from '../models';
 import { IUser, IRegisterPayload } from '~/common/interfaces';
+import { UserType, ModelAlias,  Attribute} from '~/common/enums';
+
 
 class UserRepository {
   public getAll(): Promise<IUser[]> {
     return UserModel.findAll();
   }
 
+  public getByType(type:UserType):Promise<IUser[]>{
+    if (type === UserType.DOCTOR) {
+      return UserModel.findAll({
+        where: {type},
+        include: {
+          model:DoctorModel,
+          as:ModelAlias.DOCTOR,
+          attributes: [Attribute.ID, Attribute.DEPARTMENT, Attribute.ABOUT],
+          include:[
+            {
+              model: ClinicModel,
+              as: ModelAlias.CLINIC,
+              attributes: [Attribute.ID, Attribute.NAME, Attribute.ADDRESS, Attribute.CLINIC_TYPE]
```

Comment:
It isn't attribute
You need to do something like this 
```
enum DoctorKey {
  ID = 'id,
}

interface IDoctor {
  [DoctorKey.ID]: string;
}
```

and use the `DoctorKey` enum here